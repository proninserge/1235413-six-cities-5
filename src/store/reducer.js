import offers from '@/mocks/offers';
import reviews from '@/mocks/reviews';
import {ActionType} from '@store/action';
import {getOffers} from '@/location';

const allCities = new Set([...offers.map((offer) => offer.city)]);
const cities = [...allCities];
const initialCity = cities[0];
const initialOffers = offers.filter((offer) => offer.city === initialCity);

const initialState = {
  currentCity: initialCity,
  offersForCity: initialOffers,
  cities,
  offers,
  reviews,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CITY_CHANGE:
      return Object.assign({}, state, {
        currentCity: action.city
      });
    case ActionType.GET_OFFERS:
      return Object.assign({}, state, {
        offersForCity: getOffers(state.offers, state.currentCity),
      });
    default: return state;
  }
};


export {reducer};
