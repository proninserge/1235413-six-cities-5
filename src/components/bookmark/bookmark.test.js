import renderer from 'react-test-renderer';
import Bookmark from './bookmark';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import browserHistory from '@/browser-history';
import {createStore} from 'redux';
import reducer from '@store/reducers/root-reducer';

const store = createStore(reducer);

const offers = [{
  hotelCity: {
    name: `Lafayette`,
    location: {
      lat: 30,
      lng: 19,
      zoom: 16,
    },
  },
  previewImage: `mock`,
  images: [`mock`, `mock`],
  title: `mock`,
  isFavorite: true,
  isPremium: true,
  rating: 5,
  type: `hotel`,
  bedrooms: 2,
  maxAdults: 2,
  price: 22,
  goods: [`mock`, `mock`],
  hotelHost: {
    id: 2,
    name: `John Doe`,
    isPro: true,
    avatarUrl: `mock`,
  },
  description: `Mock. Mock.`,
  hotelLocation: {
    lat: 50,
    lng: 50,
    zoom: 16,
  },
  id: 1,
},
{
  hotelCity: {
    name: `Lafayette`,
    location: {
      lat: 30,
      lng: 19,
      zoom: 16,
    },
  },
  previewImage: `mock`,
  images: [`mock`, `mock`],
  title: `mock`,
  isFavorite: true,
  isPremium: true,
  rating: 5,
  type: `hotel`,
  bedrooms: 2,
  maxAdults: 2,
  price: 22,
  goods: [`mock`, `mock`],
  hotelHost: {
    id: 2,
    name: `John Doe`,
    isPro: true,
    avatarUrl: `mock`,
  },
  description: `Mock. Mock.`,
  hotelLocation: {
    lat: 50,
    lng: 50,
    zoom: 16,
  },
  id: 2,
}];

const noop = () => {};

describe(`<Bookmark /> render`, () => {

  it(`correctly renders Bookmark if authorized`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={browserHistory}>
              <Bookmark
                className={`main`}
                onBookmarkClick={noop}
                offer={offers[0]}
                favoriteOffers={offers}
                isAuthorized={true}
              />
            </Router>
          </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`correctly renders Bookmark if not authorized`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={browserHistory}>
              <Bookmark
                className={`main`}
                onBookmarkClick={noop}
                offer={offers[0]}
                favoriteOffers={offers}
                isAuthorized={false}
              />
            </Router>
          </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});

