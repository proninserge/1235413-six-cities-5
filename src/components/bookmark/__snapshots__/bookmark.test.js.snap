// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bookmark /> render correctly renders Bookmark if authorized 1`] = `
<button
  className="main-button button main-button--active"
  onClick={[Function]}
  type="button"
>
  <svg
    className="main-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`<Bookmark /> render correctly renders Bookmark if not authorized 1`] = `
<a
  href="/login"
  onClick={[Function]}
>
  <button
    className="main-button button main-button--active"
    type="button"
  >
    <svg
      className="main-icon"
      height="19"
      width="18"
    >
      <use
        xlinkHref="#icon-bookmark"
      />
    </svg>
    <span
      className="visually-hidden"
    >
      In bookmarks
    </span>
  </button>
</a>
`;
